pkg/common/cns-lib/volume/manager.go

```diff
@@ -200,6 +208,24 @@ func ClearTaskInfoObjects() {
 				}()
 			}
 		}
+		// clear task info objects for CreateSnapshot task
+		for snapshotTaskKey, taskDetails := range snapshotTaskMap {
+			// Get the time difference between current time and the expiration
```

Comment:
Iterating over a map without picking it's lock seems like risky business to me.

Reply:
I think it is OK. Even if the current thread reads a stale snapshotTaskMap object, the worst case is we missed cleaning up some entries. It is OK to clean it up in the next round. For the task map, we are good as long as we update it in a thread-safe way.